Userland/Applications/PixelPaint/Image.cpp

```diff
@@ -555,4 +556,14 @@ void Image::set_path(String path)
     set_title(LexicalPath::basename(m_path));
 }
 
+Color Image::color_at(Gfx::IntPoint const& point) const
+{
+    Color color;
+    for (auto& layer : m_layers) {
+        if (layer.is_visible() && layer.rect().contains(point))
+            color = color.blend(layer.bitmap().get_pixel(point));
+    }
+    return color;
+}
+
```

Comment:
Would it make sense to include opacity information for the layers as well? That way we get the actual shown color

Reply:
Ah yes, you're right, I missed the layer opacity completely.